From 5706518fb5d4699ed638b38399da55962c209f3c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:46 -0400
Subject: [powerpc] Use ppc_function_entry instead of open coding it

Message-id: <1410545655-205645-418-git-send-email-dzickus@redhat.com>
Patchwork-id: 94614
O-Subject: [RHEL7 PATCH 417/626] powerpc: Use ppc_function_entry instead of open coding it
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 26f920605680b69e484a114b3dcb47ce11df9827
Author: Anton Blanchard <anton@samba.org>
Date:   Mon Mar 10 09:40:26 2014 +1100

    powerpc: Use ppc_function_entry instead of open coding it

    Replace FUNCTION_TEXT with ppc_function_entry which can handle both
    ABIv1 and ABIv2.

    Signed-off-by: Anton Blanchard <anton@samba.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/mm/hash_utils_64.c b/arch/powerpc/mm/hash_utils_64.c
index 6650699..f885e51 100644
--- a/arch/powerpc/mm/hash_utils_64.c
+++ b/arch/powerpc/mm/hash_utils_64.c
@@ -603,8 +603,6 @@ int remove_section_mapping(unsigned long start, unsigned long end)
 }
 #endif /* CONFIG_MEMORY_HOTPLUG */
 
-#define FUNCTION_TEXT(A) ((*(unsigned long *)(A)))
-
 static void __init htab_finish_init(void)
 {
  extern unsigned int *htab_call_hpte_insert1;
@@ -619,31 +617,31 @@ static void __init htab_finish_init(void)
  extern unsigned int *ht64_call_hpte_updatepp;
 
  patch_branch(ht64_call_hpte_insert1,
-  FUNCTION_TEXT(ppc_md.hpte_insert),
+  ppc_function_entry(ppc_md.hpte_insert),
   BRANCH_SET_LINK);
  patch_branch(ht64_call_hpte_insert2,
-  FUNCTION_TEXT(ppc_md.hpte_insert),
+  ppc_function_entry(ppc_md.hpte_insert),
   BRANCH_SET_LINK);
  patch_branch(ht64_call_hpte_remove,
-  FUNCTION_TEXT(ppc_md.hpte_remove),
+  ppc_function_entry(ppc_md.hpte_remove),
   BRANCH_SET_LINK);
  patch_branch(ht64_call_hpte_updatepp,
-  FUNCTION_TEXT(ppc_md.hpte_updatepp),
+  ppc_function_entry(ppc_md.hpte_updatepp),
   BRANCH_SET_LINK);
 
 #endif /* CONFIG_PPC_HAS_HASH_64K */
 
  patch_branch(htab_call_hpte_insert1,
-  FUNCTION_TEXT(ppc_md.hpte_insert),
+  ppc_function_entry(ppc_md.hpte_insert),
   BRANCH_SET_LINK);
  patch_branch(htab_call_hpte_insert2,
-  FUNCTION_TEXT(ppc_md.hpte_insert),
+  ppc_function_entry(ppc_md.hpte_insert),
   BRANCH_SET_LINK);
  patch_branch(htab_call_hpte_remove,
-  FUNCTION_TEXT(ppc_md.hpte_remove),
+  ppc_function_entry(ppc_md.hpte_remove),
   BRANCH_SET_LINK);
  patch_branch(htab_call_hpte_updatepp,
-  FUNCTION_TEXT(ppc_md.hpte_updatepp),
+  ppc_function_entry(ppc_md.hpte_updatepp),
   BRANCH_SET_LINK);
 }
 
-- 
1.7.1